src/aosm/azext_aosm/build_processors/nexus_image_processor.py

```diff
@@ -0,0 +1,171 @@
+    """
+
+    def __init__(self, name: str, input_artifact: NexusImageFileInput):
+        super().__init__(name, input_artifact)
+        self.input_artifact: NexusImageFileInput = input_artifact
+
+    def get_artifact_manifest_list(self) -> List[ManifestArtifactFormat]:
+        """
+        Get the list of artifacts for the artifact manifest.
+
+        :return: A list of artifacts for the artifact manifest.
+        :rtype: List[ManifestArtifactFormat]
+        """
+        logger.info("Getting artifact manifest list for Nexus image input.")
+        return [
+            ManifestArtifactFormat(
+                artifact_name=self.input_artifact.artifact_name,
+                artifact_type=ArtifactType.IMAGE_FILE.value,
+                artifact_version=self.input_artifact.artifact_version,
+            )
+        ]
+
+    def get_artifact_details(
+        self,
+    ) -> Tuple[List[BaseArtifact], List[LocalFileBuilder]]:
+        """
+        Get the artifact details for publishing.
+
+        :return: A tuple containing the list of artifacts and the list of local file builders.
+        :rtype: Tuple[List[BaseArtifact], List[LocalFileBuilder]]
+        """
+        logger.info("Getting artifact details for Nexus image input.")
+        artifacts: List[BaseArtifact] = []
+        file_builders: List[LocalFileBuilder] = []
+
+        # We only support remote ACR artifacts for container images
+        artifacts.append(
+            RemoteACRArtifact(
+                artifact_name=self.input_artifact.artifact_name,
+                artifact_type=ArtifactType.IMAGE_FILE.value,
+                artifact_version=self.input_artifact.artifact_version,
+                source_registry=self.input_artifact.source_acr_registry,
+                source_registry_namespace="",
+            )
+        )
+        return artifacts, file_builders
+
+    def generate_nf_application(self) -> AzureOperatorNexusNetworkFunctionImageApplication:
+        """
+        Generate the NF application.
+
+        :return: The NF application.
+        :rtype: AzureOperatorNexusNetworkFunctionImageApplication
+        """
+        logger.info("Generating NF application for Nexus image input.")
+
+        return AzureOperatorNexusNetworkFunctionImageApplication(
+            name=self.name,
+            depends_on_profile=DependsOnProfile(install_depends_on=[],
+                                                uninstall_depends_on=[], update_depends_on=[]),
+            artifact_profile=self._generate_artifact_profile(),
+            deploy_parameters_mapping_rule_profile=self._generate_mapping_rule_profile(),
+        )
+
+    def generate_resource_element_template(self) -> ResourceElementTemplate:
+        """
+        Generate the resource element template.
+
+        :raises NotImplementedError: NSDs do not support deployment of Nexus images.
+        """
+        raise NotImplementedError("NSDs do not support deployment of Nexus images.")
```

Comment:
I don't understand this. Copy/paste error?

Also in VHDProcessor, but I can't comment there because it hasn't changed. (Why GitHub, why?!)